tests/agents/system/test_disk.py

```diff
@@ -0,0 +1,139 @@
+        )
+        disk_fill_agent = DiskFill(disk_fill_config)
+
+        self.assertEqual(
+            str(Path().resolve()), disk_fill_config.partition.absolute().as_posix()
+        )
+        self.assertEqual(50, disk_fill_config.partition_pct)
+
+        disk_fill_agent.setup()
+        self.assertEqual(AgentState.SETUP, disk_fill_agent.current_state)
+
+        disk_fill_agent.run()
+        self.assertEqual(AgentState.RUNNING, disk_fill_agent.current_state)
+
+        temp_dir_path = Path().resolve() / "temp"
+        self.assertTrue(temp_dir_path.exists())
+        disk_dir = Path().resolve() / "temp/filler0.txt"
+        self.assertEqual(512, disk_dir.stat().st_size)
+
+        disk_fill_agent.teardown()
+        self.assertEqual(AgentState.TEARDOWN, disk_fill_agent.current_state)
+        self.assertFalse(temp_dir_path.exists())
+
+    def test_disk_fill_when_percentage_fill_full(self):
+        usage = collections.namedtuple("usage", "total used free")
+        stats = usage(2048, 1024, 1024)
+        when(shutil).disk_usage(Path(".")).thenReturn(stats)
+        disk_fill_config = DiskFillConfig(
+            duration=0.1, partition="./", partition_pct=100
+        )
+        disk_fill_agent = DiskFill(disk_fill_config)
+
+        self.assertEqual(
+            str(Path().resolve()), disk_fill_config.partition.absolute().as_posix()
+        )
+        self.assertEqual(100, disk_fill_config.partition_pct)
+
+        disk_fill_agent.setup()
+        self.assertEqual(AgentState.SETUP, disk_fill_agent.current_state)
+
+        disk_fill_agent.run()
+        self.assertEqual(AgentState.RUNNING, disk_fill_agent.current_state)
+
+        disk_dir = Path().resolve() / "temp/filler0.txt"
+        self.assertEqual(1024, disk_dir.stat().st_size)
+
+        disk_fill_agent.teardown()
+        self.assertEqual(AgentState.TEARDOWN, disk_fill_agent.current_state)
+
+    def test_disk_fill_when_percentage_fill_stop_run(self):
+        disk_fill_config = DiskFillConfig(
+            duration=0.1, partition="./", partition_pct=100
+        )
+        disk_fill_agent = DiskFill(disk_fill_config)
+
+        self.assertEqual(
+            str(Path().resolve()), disk_fill_config.partition.absolute().as_posix()
+        )
+        self.assertEqual(100, disk_fill_config.partition_pct)
+
+        disk_fill_agent.setup()
+        self.assertEqual(AgentState.SETUP, disk_fill_agent.current_state)
+
+        disk_fill_agent.stop_async_run = True
+
+        disk_fill_agent.run()
+        self.assertEqual(AgentState.RUNNING, disk_fill_agent.current_state)
+
+        temp_file_0 = Path().resolve() / "temp/filler0.txt"
+        self.assertFalse(temp_file_0.exists())
+
+        disk_fill_agent.teardown()
+        self.assertEqual(AgentState.TEARDOWN, disk_fill_agent.current_state)
+
+    def test_disk_teardown_without_disk_fill_dir(self):
+        usage = collections.namedtuple("usage", "total used free")
+        stats = usage(4096, 2048, 2048)
```

Comment:
```suggestion
        stats = usage(4096, 4096, 0)
```

With this usage value, run should never create the directory. This is a better check for this case.